internal/bminventory/inventory.go

```diff
@@ -5838,8 +5857,23 @@ func (b *bareMetalInventory) V2DownloadInfraEnvFiles(ctx context.Context, params
 		}
 		filename = fmt.Sprintf("%s-%s", params.InfraEnvID, params.FileName)
 	case "static-network-config":
+		var netFiles []staticnetworkconfig.StaticNetworkConfigData
 		if infraEnv.StaticNetworkConfig != "" {
-			netFiles, err := b.staticNetworkConfig.GenerateStaticNetworkConfigData(ctx, infraEnv.StaticNetworkConfig)
+			// backward compatibility - nmstate.service has been available on RHCOS since version 4.13+, therefore, we should maintain both flows
+			ocpVersion, err := version.NewVersion(infraEnv.OpenshiftVersion)
+			if err != nil {
+				return common.GenerateErrorResponder(errors.Errorf("Failed to parse OCP version"))
+			}
+			minimalVersionForNmstatectl, err := version.NewVersion(constants.MinimalVersionForNmstatectl)
+			if err != nil {
+				return common.GenerateErrorResponder(errors.Errorf("Failed to parse minimal ocp version for nmstatectl"))
+			}
+			// TODO: Remove the architecture condition after fetching the nmstatectl binary from rootfs.
+			if ocpVersion.GreaterThanOrEqual(minimalVersionForNmstatectl) && infraEnv.CPUArchitecture == "x86_64" {
```

Comment:
Since this is being used multiple times lets put it in a helper function in the `staticnetworkconfig` package.

Then we have a good home for the version constant as well.

Reply:
After updating with the VersionGreaterOrEqual you suggested:
`		// backward compatibility - nmstate.service has been available on RHCOS since version 4.13+, therefore, we should maintain both flows
		ok, err := common.VersionGreaterOrEqual(infraEnv.OpenshiftVersion, staticnetworkconfig.MinimalVersionForNmstatectl)
		if err != nil {
			return "", err
		}

		// TODO: Remove the architecture condition after fetching the nmstatectl binary from rootfs.
		if ok && infraEnv.CPUArchitecture == common.X86CPUArchitecture {`
		
I don't see a need for a helper function. what do you think?